outputs.tf

```diff
@@ -0,0 +1,87 @@
+output "public_ips" {
+  description = "List of Public IPs of instances (or EIP)"
+  value       = "${distinct(compact(concat(aws_eip.default.*.public_ip,aws_instance.default.*.public_ip, aws_eip.additional.*.public_ip, list() )))}"
+}
+
+output "private_ip" {
+  description = "Private IPs of instances"
+  value       = "${split(",",join(",", aws_instance.default.*.private_ip))}"
+}
+
+output "private_dns" {
+  description = "Private DNS records of instances"
+  value       = "${split(",",join(",", aws_instance.default.*.private_dns))}"
+}
+
+locals {
+  ip_dns_list = "${split(",", replace(join(",", distinct(compact(concat(aws_eip.default.*.public_ip,aws_instance.default.*.public_ip, aws_eip.additional.*.public_ip, list() )))), ".", "-"))}"
+  dns_names   = "${formatlist("%v.${var.region == "us-east-1" ? "compute-1" : "${var.region}.compute"}.amazonaws.com", distinct(compact(local.ip_dns_list)))}"
+}
+
+output "public_dns" {
+  value = "${local.dns_names}"
+}
+
+output "id" {
+  description = "Disambiguated ID list"
+  value       = "${aws_instance.default.*.id}"
+}
+
+output "aws_key_pair_name" {
+  description = "Name of AWS key pair"
+  value       = "${signum(length(var.ssh_key_pair)) == 1 ? var.ssh_key_pair : "${var.generate_ssh_key_pair == "true" ? module.ssh_key_pair.key_name : ""}"}"
+}
+
+output "new_ssh_keypair_generated" {
+  value = "${signum(length(var.ssh_key_pair)) == 1 ? "false" : "true" }"
+}
+
+output "ssh_key_pem_path" {
+  description = "Path where SSH key pair was created (if applicable)"
+  value       = "${local.ssh_key_pair_path}/${module.ssh_key_pair.key_name}.pem"
+}
+
+output "security_group_ids" {
+  description = "ID on the new AWS Security Group associated with creating instance"
+  value       = "${compact(concat(list(var.create_default_security_group == "true" ? join("", aws_security_group.default.*.id) : ""), var.security_groups))}"
+}
+
+output "role" {
+  description = "Name of AWS IAM Role associated with creating instance"
+  value       = "${join(",", aws_iam_role.default.*.name)}"
```

Comment:
I think we should return a `list` for all outputs since this module is for N servers